mtp_api/apps/credit/dump.py

```diff
@@ -0,0 +1,159 @@
+        queryset = Credit.objects_all \
+            .exclude(resolution=CREDIT_RESOLUTION.INITIAL) \
+            .exclude(payment__status=PAYMENT_STATUS.EXPIRED)
+        if self.only_with_triggered_rules:
+            queryset = queryset \
+                .exclude(security_check__rules__len=0) \
+                .exclude(security_check__rules__isnull=True)
+        return queryset
+
+    def get_headers(self):
+        headers = super().get_headers() + [
+            'URL',
+            'Date started', 'Date received', 'Date credited',
+            'Amount',
+            'Status',
+
+            'Prisoner number', 'Prisoner name', 'Prison',
+
+            'Payment method',
+            'Sender name',
+        ]
+        if not self.only_with_triggered_rules:
+            # if only credits that triggered security rules are included, there can be no bank transfers
+            headers += [
+                'Bank transfer sort code', 'Bank transfer account', 'Bank transfer roll number',
+            ]
+        headers += [
+            'Debit card first six digits', 'Debit card last four digits', 'Debit card expiry',
+            'Debit card billing address line 1', 'Debit card billing address line 2', 'Debit card billing address city',
+            'Debit card billing address postcode', 'Debit card billing address country',
+            'Sender email', 'Sender IP address',
+
+            'Owner username', 'Blocked',  # TODO: these might be removed, see below
+
+            'NOMIS transaction', 'WorldPay order code',
+
+            'Security check codes', 'Security check description',
+            'Security check status', 'Security check actioned by', 'Security check rejection reasons',
+        ]
+        return headers
+
+    def serialise(self, record: Credit):
+        status = record.status
+        if status:
+            status = CREDIT_STATUS.for_value(status).display
+        else:
+            status = 'Anonymous'
+
+        if hasattr(record, 'security_check'):
+            security_check = record.security_check
+            security_check_description = '; '.join(security_check.description)
+            security_check_status = CHECK_STATUS.for_value(security_check.status).display
+            security_check_actioned_by = security_check.actioned_by.username if security_check.actioned_by else ''
+            if len(security_check.rules) > 0:
+                security_check_rules = security_check.rules
+                security_check_rejection_reasons = '; '.join(
+                    human_readable_check_rejection_reasons(security_check.rejection_reasons)
+                )
+            else:
+                security_check_rules = None
+                security_check_rejection_reasons = None
+        else:
+            security_check_description = None
+            security_check_status = None
+            security_check_actioned_by = None
+            security_check_rules = None
+            security_check_rejection_reasons = None
```

Comment:
I wonder if logic to serialise the `security_check` could be swept in a `serialise_security_check()` method similarly to what it's already done with the `serialise_sender()` method.